luaa.c

```diff
@@ -411,6 +412,126 @@ get_modifier_name(int map_index)
     return 0; /* \0 */
 }
 
+// Helper function for luaA_get_key_name() below.
+// Will return the UTF-32 codepoint IF AND ONLY IF the input is exactly one
+// valid UTF-8 character. Otherwise, it will return zero.
+static uint32_t
+one_utf8_to_utf32(const char* input, const size_t length) {
+    // When casting a char to a uint32_t, the missing bits will all be set to
+    // the highest bit in the original char. Thus the exaggerated masking of
+    // every value (Unicode is defined around bytes)
+    if (length < 1 || length > 4)
+        return 0;
+    if (length == 1)
+    {
+        if ((*input & 0x80) == 0)
+            return *input;
+        else
+            return 0;
+    }
+    uint32_t mask = (0xff << (7 - length)) & 0x000000ff;
+    const uint32_t check = (mask << 1) & 0x000000ff;
+    uint32_t ucs = 0x00000000;
+    if ((*input & mask) == check)
+    {
+        mask = (~mask) & 0x000000ff;
+        ucs = (*input & mask) << (6 * (length-1));
+    }
+    else
+        return 0;
+    for (size_t i = 1; i < length; i++)
+    {
+        if ((*(input + i) & 0x000000c0) != 0x00000080)
+            return 0;
+        ucs += (*(input + i) & 0x0000003f) << (6 * (length-i-1));
+    }
+    return ucs;
+}
+
+/** Get Xlib keysym and xkb_keysym_to_uf8 from an Awesome keycode.
```

Comment:
```suggestion
/** Get X11 keysym and xkb_keysym_to_uf8 from an Awesome keycode.
```
No idea what that second part with `xkb_keysym_to_uf8` is supposed to mean. Can we perhaps call this something understandable? `A string representation of the key`?

Reply:
Yes, please. It's not even correctly written, it should be `xkb_keysym_to_utf8`. (I forgot the T.)

`A string representation of the key` is pretty legible, but if you can write a clearer expression, please tell me.